tests/message_processing/test_process_instances.py

```diff
@@ -2,6 +2,7 @@
 import itertools
```

Comment:
I 'd like to see more tests: 
* unit tests for the code that computes the cost of a single InstanceContent object.
* a test here that tries to process an instance without having enough tokens.

Reply:
Where are the additional tests?